### ISSUE TITLE
Table of contents for paper

### ISSUE THREAD

**Issue by AnneCarpenter** [greenelab/deep-review#957]
It'd be lovely to generate a table of contents for the full paper, so people can see an overview of the topic!

**Comment by evancofer** [comment]
@AnneCarpenter I agree! @dhimmel @cgreene is this something that can be easily added with Manubot?

**Comment by dhimmel** [comment]
The [HTML view](https://greenelab.github.io/deep-review/v/ec05a06890580987fd84eeadaaa8422bda2f918f/) has a slide-out table of contents:

> ![image](https://user-images.githubusercontent.com/1117703/59964655-5c5a2c80-94d1-11e9-974f-352194cfc2a4.png)

If you update to the latest Manubot, the default behavior of the interactive HTML TOC will be a bit more intuitive. @vincerubinetti the deep review seems to be an example where the TOC headings are a bit large in the side bar?

@AnneCarpenter and @evancofer is this what you're looking for? If so, there is an option to make the side bar TOC always open, if that would help make users aware of it's existence.

If you would like a TOC to be included in the actual document as a section, including in the PDF, then you could try adding Pandoc's `--table-of-contents` argument in `build.sh`. Haven't ever used this, but would be interested in seeing it.

**Comment by AnneCarpenter** [comment]
That solves my problem!! Indeed, setting the sidebar as default would avoid people like me missing it (the downside would be that sometimes folks are annoyed when there is a lot of clutter on the page and they just want to see the content!)

**Comment by agitter** [comment]
Thanks for the suggestion @AnneCarpenter !

I haven't tried adding `--table-of-contents` to the PDF build like @dhimmel suggested, but the current PDF does show bookmarks correctly as an alternative overview:
![image](https://user-images.githubusercontent.com/3532898/59968248-604f7400-94fc-11e9-9d21-b17cf768fbc9.png)

**Comment by evancofer** [comment]
@dhimmel Yes, I think it looks good. I agree with @AnneCarpenter that the expanded-by-default behavior could be nice, but I do think it could get in the way on smaller mobile screens. That being said, I think a paper of this size may be destined for reading difficulties on mobile viewers.

**Comment by dhimmel** [comment]
> the current PDF does show bookmarks correctly as an alternative overview

This will actually be ruined by https://github.com/greenelab/deep-review/pull/958 which will switch to using athenapdf rather than weasyprint to generate the PDF. Athena doesn't have the bookmark/outline feature as per https://github.com/arachnys/athenapdf/issues/16#issuecomment-505114680.

For the HTML view, we may be able to play with the plugin configuration to have it open by default:

https://github.com/manubot/rootstock/blob/aa09b789f53d0deea89c6d6ca563061c57bfed5b/build/plugins/table-of-contents.html#L25-L28

**Comment by vincerubinetti** [comment]
> Actually, it seems that there is no start-open option, @vincerubinetti?

In the current state of the repo, you're right. It always starts closed, but clicking on a TOC link or clicking the page when the TOC is open will respect the `open` option. We'll revisit this behavior soon and decide what to do.
